Library/ViewModels/DiscoveryViewModel.swift

```diff
@@ -122,6 +122,11 @@ public final class DiscoveryViewModel: DiscoveryViewModelType, DiscoveryViewMode
     self.configureNavigationHeader = dataSourceConfiguredAndCurrentParams
     self.loadFilterIntoDataSource = dataSourceConfiguredAndCurrentParams
 
+    let prevSortedPage = Signal.merge(
+      self.previouslySortPagerSelectedSortProperty.signal,
+      self.previouslyTransitionedToPageProperty.signal.map { sorts[$0] }
+    ).take(first: 1)
```

Comment:
Could you clarify why you are only taking the first here?

Reply:
Because, there are two ways to update `prevSortedPage`, I only need the first one that emit at on page transition.